prov/verbs/src/fi_verbs.c

```diff
@@ -872,6 +881,8 @@ fi_ibv_msg_ep_sendmsg(struct fid_ep *ep, const struct fi_msg *msg, uint64_t flag
 	} else {
 		wr.send_flags = 0;
 	}
+	if ((_ep->ep_flags & flags & FI_COMPLETION))
+		wr.send_flags |= IBV_SEND_SIGNALED;
```

Comment:
If ep_flags is not set then we have to generate completion by default. Same comment applies for writemsg, readmsg and atomics below.